cmd/ui/src/views/Users/Users.tsx

```diff
@@ -159,8 +160,19 @@ const Users = () => {
     });
 
     const updateUserPasswordMutation = useMutation(
-        ({ userId, secret, needsPasswordReset }: { userId: string; secret: string; needsPasswordReset: boolean }) =>
+        ({
+            userId,
+            secret,
+            needsPasswordReset,
+            currentSecret,
+        }: {
+            currentSecret?: string;
+            userId: string;
+            secret: string;
+            needsPasswordReset: boolean;
+        }) =>
             apiClient.putUserAuthSecret(userId, {
+                ...(currentSecret && { current_secret: currentSecret }),
```

Comment:
update the types.ts file in js-client-library to include the new current_secret field. the interface you are looking for is called `PutUserAuthSecretRequest`